packages/patternfly-4/react-core/src/components/OptionsMenu/OptionsMenuItem.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, getModifier } from '@patternfly/react-styles';
+import styles from '@patternfly/patternfly/components/OptionsMenu/options-menu.css';
+import { CheckIcon } from '@patternfly/react-icons';
+import { KEY_CODES, KEYHANDLER_DIRECTION } from '../../helpers';
+
+const propTypes = {
+  /** Anything which can be rendered as an Options menu item */
+  children: PropTypes.node,
+  /** Classes applied to root element of an Options menu item */
+  className: PropTypes.string,
+  /** Render Options menu item as selected */
+  isSelected: PropTypes.bool,
+  /** Render Options menu item as disabled option */
+  isDisabled: PropTypes.bool,
+  /** Callback for when this Options menu item is selected */
+  onSelect: PropTypes.func,
+  /** Unique id of this Options menu item */
+  id: PropTypes.string,
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  isSelected: false,
+  isDisabled: false,
+  onSelect: Function.prototype,
+  id: '',
+};
+
+class OptionsMenuItem extends React.Component {
+
+  onKeyDown = event => {
+    // Detected key press on this item, notify the menu parent so that the appropriate
+    // item can be focused
+    if (event.keyCode === KEY_CODES.TAB) return;
+    event.preventDefault();
+    if (event.keyCode === KEY_CODES.ARROW_UP) {
+      this.props.context.keyHandler(this.props.index, KEYHANDLER_DIRECTION.UP);
+    } else if (event.keyCode === KEY_CODES.ARROW_DOWN) {
+      this.props.context.keyHandler(this.props.index, KEYHANDLER_DIRECTION.DOWN);
+    } else if (event.keyCode === KEY_CODES.ENTER) {
+      this.props.onSelect(event);
+    }
+  };
+
+  render() {
+    const { onSelect, isDisabled, isSelected, className, children, id, ...props } = this.props;
+    return <button
+      className={css(styles.optionsMenuMenuItem, isDisabled && getModifier(styles, 'disabled'), className)}
+      aria-disabled={isDisabled}
+      onClick={onSelect}
+      onKeyDown={this.onKeyDown}
+      aria-selected={isSelected}
+      id={id}
+      {...props}>
+      {children}
+      <i className={css(styles.optionsMenuMenuItemIcon)} aria-hidden={true} hidden={!isSelected}><CheckIcon/></i>
```

Comment:
there is no need to wrap the icon in `<i>`.  You can apply the styles and attributes directly to the CheckIcon

Reply:
It doesn't seem to work unless it's in <i> at the moment. After working with Dana and Sarah on trying to determine why, we decided to open an issue for this rather that delay this PR.